lib/ansible/modules/packaging/os/yum_repository.py

```diff
@@ -424,10 +426,11 @@
     mirrorlist: http://mirrorlist.repoforge.org/el7/mirrors-rpmforge
     enabled: no
 
-- name: Remove repository
+- name: Remove repository (and clean up left-over metadata)
   yum_repository:
     name: epel
     state: absent
+  notify: yum-clean-all
```

Comment:
Example of such handler should be provided. This is what I normally user as the handler:

```
- name: Clean YUM cache
  shell: env yum clean all
```

The `env` is there to prevent Ansible from displaying warning that the `yum` module should be used instead. The same can be used for `service` commands (e.g `env service postgresql init`).